extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -138,24 +138,25 @@ class AmpViqeoPlayer extends AMP.BaseElement {
       {
         allowFullscreen: true,
       }
-    );
-    iframe.title = this.element.title || 'Viqeo video';
+    ).then((iframe) => {
+      iframe.title = this.element.title || 'Viqeo video';
 
-    // required to display the user gesture in the iframe
-    iframe.setAttribute('allow', 'autoplay');
+      // required to display the user gesture in the iframe
+      iframe.setAttribute('allow', 'autoplay');
 
-    this.unlistenMessage_ = listen(
-      this.win,
-      'message',
-      this.handleViqeoMessages_.bind(this)
-    );
+      this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleViqeoMessages_.bind(this)
+      );
 
-    return this.mutateElement(() => {
-      this.element.appendChild(iframe);
-      this.iframe_ = iframe;
-      this.applyFillContent(iframe);
-    }).then(() => {
-      return this.playerReadyPromise_;
+      return this.mutateElement(() => {
+        this.element.appendChild(iframe);
+        this.iframe_ = iframe;
+        this.applyFillContent(iframe);
+      }).then(() => {
+        return this.playerReadyPromise_;
+      });
```

Comment:
nit: don't nest promise chain.
```suggestion
      return this.mutateElement(() => {
        this.element.appendChild(iframe);
        this.iframe_ = iframe;
        this.applyFillContent(iframe);
      });
    })
    .then(() => this.playerReadyPromise_);
```